src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -255,6 +255,12 @@ fn get_host_info() -> Result<HostInfo> {
 
     let guest_protection = guest_protection.to_string();
 
+    let mut vm_container_capable = true;
+
+    if arch_specific::host_is_vmcontainer_capable().is_err() {
+        vm_container_capable = false;
+    }
```

Comment:
Testing this PR in a VM that can run a container, `env` shows `vm_container_capable = false` for me. Adding some debug shows the problem is:

```
check_kernel_modules error: Err(You are running in a VM, where the kernel module 'kvm_intel' parameter 'unrestricted_guest' has a value 'Y'. This causes conflict when running kata.)
```

Reply:
ftr, the QEMU VM I'm running `kata-ctl` inside of is created like this:

```bash
$ qemu-system-x86_64 -hda ./ubuntu-22.04.qcow2 -name ubuntu-22.04.qcow2 -smp cpus=4 -m 8192 -cpu host,host-phys-bits -enable-kvm -net nic,model=virtio -net user,hostfwd=tcp::1826-:22,hostfwd=tcp::1679-:16686 -device virtio-rng-pci -qmp unix:/tmp/qmp.sock,server,nowait -nographic -serial mon:stdio -device vhost-vsock-pci,id=vhost-vsock-pci0,guest-cid=3
```

Reply:
@jodh-intel I have added a fix for the above issue so that it is safe to ignore the `unrestricted_guest` parameter while running in a VM.